### PR TITLE
build: add apple silicon support

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Does what it says on the tin, the patch surface area has been reduced to basically 0 as a result of upstream work.  There are still a few things being upstreamed and the remaining patches should disappear then.

Notes: Added new experimental apple silicon (darwin arm64) builds

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1939,22 +1957,14 @@ jobs:
...
       <<: *env-testing-build
     <<: *steps-electron-gn-check
 
-  osx-chromedriver:
-    <<: *machine-mac
-    environment:
-      <<: *env-machine-mac
-      <<: *env-release-build
-      <<: *env-send-slack-notifications
-    <<: *steps-chromedriver-build
```
Comment (nornagon) [inline .circleci/config.yml:1955]: Was it intentional to remove the chromedriver build?

File: .circleci/config.yml [hunk 2]
```diff
@@ -2009,7 +2036,7 @@ jobs:
           checkout-and-assume-cache: true
           attach: false
 
-  mas-testing-gn-check:
+  mas-testing-x64-gn-check:
```
Comment (nornagon) [inline .circleci/config.yml:2039]: do we need to build MAS versions of arm64 as well?

File: patches/chromium/chore_disable_failing_blink_font_dcheck_on_big_sur.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+   SkFontParameters::Variation::Axis axes_parameters[num_axes];
+-  int returned_axes =
+-      typeface->getVariationDesignParameters(axes_parameters, num_axes);
+-  DCHECK_EQ(num_axes, returned_axes);
+-  DCHECK_GE(num_axes, 0);
++  // int returned_axes =
++  //     typeface->getVariationDesignParameters(axes_parameters, num_axes);
++  // DCHECK_EQ(num_axes, returned_axes);
++  // DCHECK_GE(num_axes, 0);
+
```
Comment (nornagon) [inline patches/chromium/chore_disable_failing_blink_font_dcheck_on_big_sur.patch:24]: This doesn't seem _super_ upstreamable. When will we know when we can remove this?

File: patches/chromium/fix_check_issecureeventinputenabled_in_constructor_before_setting.patch [hunk 4]
```diff
@@ -0,0 +1,21 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: John Kleinschmidt <jkleinsc@github.com>
+Date: Thu, 9 Jul 2020 11:16:30 -0400
+Subject: fix: check IsSecureEventInputEnabled in constructor before setting
+ SetPasswordInputEnabled to true
+
+It appears that macOS 12 has IsSecureEventInputEnabled set to true initially.
```
Comment (nornagon) [inline patches/chromium/fix_check_issecureeventinputenabled_in_constructor_before_setting.patch:7]: this is upstreamed already, right?

File: patches/v8/chore_add_v8_apple_silicon_patches.patch [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+--- a/src/codegen/arm64/macro-assembler-arm64.cc
++++ b/src/codegen/arm64/macro-assembler-arm64.cc
+@@ -2973,6 +2973,35 @@ void TurboAssembler::PrintfNoPreserve(const char* format,
+ 
+   int arg_count = kPrintfMaxArgCount;
+ 
++#if V8_OS_MACOSX && !USE_SIMULATOR
++  CPURegList tmp_list = kCallerSaved;
++  tmp_list.Remove(x0);      // Used to pass the format string.
++  tmp_list.Remove(arg0, arg1, arg2, arg3);
```
Comment (nornagon) [inline patches/v8/chore_add_v8_apple_silicon_patches.patch:19]: extremely terrifying patch

File: script/spec-runner.js [hunk 6]
```diff
@@ -225,6 +225,9 @@ async function installSpecModules (dir) {
     npm_config_nodedir: nodeDir,
     npm_config_msvs_version: '2019'
   });
+  if (fs.existsSync(path.resolve(dir, 'node_modules'))) {
+    await fs.remove(path.resolve(dir, 'node_modules'));
+  }
```
Comment (nornagon) [inline script/spec-runner.js:230]: big oof

File: patches/chromium/fix_check_issecureeventinputenabled_in_constructor_before_setting.patch (hunk 4 above)
Comment (jkleinsc) [reply patches/chromium/fix_check_issecureeventinputenabled_in_constructor_before_setting.patch:7]: Yes - https://chromium-review.googlesource.com/c/chromium/src/+/2295749

File: .circleci/config.yml (hunk 1 above)
Comment (MarshallOfSound) [reply .circleci/config.yml:1955]: This build was unused and completely unreferenced

File: .circleci/config.yml (hunk 2 above)
Comment (MarshallOfSound) [reply .circleci/config.yml:2039]: Yes but not right now

File: patches/chromium/fix_check_issecureeventinputenabled_in_constructor_before_setting.patch (hunk 4 above)
Comment (MarshallOfSound) [reply patches/chromium/fix_check_issecureeventinputenabled_in_constructor_before_setting.patch:7]: Yes but we haven't absorbed it yet

File: patches/chromium/chore_disable_failing_blink_font_dcheck_on_big_sur.patch (hunk 3 above)
Comment (MarshallOfSound) [reply patches/chromium/chore_disable_failing_blink_font_dcheck_on_big_sur.patch:24]: Chrome are aware of the failure on Big Sur and are working themselves on fixing it

File: script/spec-runner.js (hunk 6 above)
Comment (MarshallOfSound) [reply script/spec-runner.js:230]: We can remove this once my `node-gyp` changes are absorbed upstream

File: patches/v8/chore_add_v8_apple_silicon_patches.patch (hunk 5 above)
Comment (MarshallOfSound) [reply patches/v8/chore_add_v8_apple_silicon_patches.patch:19]: V8 is mostly terrifying

**Pull request merged**